analysis/tests/combiners_test.py

```diff
@@ -822,308 +970,259 @@ def test_create_accumulator(self, metric_type, probability_to_keep, metric,
                  rel_error_l0_variance=0.08,
                  rel_error_variance=0.12,
-                 rel_error_quantiles=[-0.4],
-                 error_expected_w_dropped_partitions=-2.0,
-                 rel_error_expected_w_dropped_partitions=-0.4,
+                 rel_error_quantiles=[-0.2],
+                 error_expected_w_dropped_partitions=-1.0,
+                 rel_error_expected_w_dropped_partitions=-0.1,
                  noise_std=1.0,
              ),
              expected=metrics.AggregateErrorMetrics(
-                 metric_type=metrics.AggregateMetricType.SUM,
-                 ratio_data_dropped_l0=0.0,
+                 metric_type=metrics.AggregateMetricType.PRIVACY_ID_COUNT,
+                 ratio_data_dropped_l0=0.2,
                  ratio_data_dropped_linf=0.0,
                  ratio_data_dropped_partition_selection=0.0,
                  error_l0_expected=-1.0,
-                 error_linf_expected=-1.0,
-                 error_expected=-2.0,
+                 error_linf_expected=0.0,
+                 error_linf_min_expected=0.0,
+                 error_linf_max_expected=0.0,
+                 error_expected=-1.0,
                  error_l0_variance=2.0,
                  error_variance=3.0,
-                 error_quantiles=[-2.0],
+                 error_quantiles=[-1.0],
                  rel_error_l0_expected=-0.2,
-                 rel_error_linf_expected=-0.2,
-                 rel_error_expected=-0.4,
+                 rel_error_linf_expected=0.0,
+                 rel_error_linf_min_expected=0.0,
+                 rel_error_linf_max_expected=0.0,
+                 rel_error_expected=-0.2,
                  rel_error_l0_variance=0.08,
                  rel_error_variance=0.12,
-                 rel_error_quantiles=[-0.4],
-                 error_expected_w_dropped_partitions=-2.0,
-                 rel_error_expected_w_dropped_partitions=-0.4,
+                 rel_error_quantiles=[-0.2],
+                 error_expected_w_dropped_partitions=-1.0,
+                 rel_error_expected_w_dropped_partitions=-0.1,
                  noise_std=1.0,
              )),
     )
-    def test_compute_metrics(self, acc, expected):
-        combiner = combiners.SumAggregateErrorMetricsCombiner([0.5])
+    def test_compute_metrics(self, metric_type, acc, expected):
+        combiner = combiners.SumAggregateErrorMetricsCombiner(
+            metric_type, [0.5])
         metric = combiner.compute_metrics(acc)
-        self.assertEqual(expected, metric)
+        assert_dataclasses_are_equal(self, expected, metric)
 
         # Test that no type is np.float64
         self.assertTrue(_check_none_are_np_float64(acc))
 
 
+def assert_dataclasses_are_equal(test, expected, actual):
+    test.assertEquals(type(expected), type(actual))
+    expected = dataclasses.asdict(expected)
+    actual = dataclasses.asdict(actual)
+
+    for (f1, exp), (f2, act) in zip(expected.items(), actual.items()):
```

Comment:
Are there guarantees that `expected.items()` are `actual.items()` are in the order? It feels that it is not (both are hash dictionaries, and hash function might be in principle chosen randomly). I'd suggest something like

```
for field_name in actual.keys():
  expected_value = expected[...]
  actual = actual[...]
```